main.tf

```diff
@@ -1,3 +1,7 @@
+locals {
+  enable_ingress_application_gateway = var.enable_ingress_application_gateway == null ? false : var.enable_ingress_application_gateway
```

Comment:
Hi @jcorioland ,

I've removed this local, it'll prevent the user from disabling ingress app gateway, and I'm still testing it.
I think setting variable's default value to null will disable this whole ingress application gateway block for who is still using current module, when they upgrade the module version they won't be surprised by an update-in-place.

Reply:
I've removed the local, would you please give it another review? Thanks!

Reply:
I was suggesting to set a default value to `false` so anybody who get the last version of the module but don't pay attention to the new variable have it set to false by default, without doing anything. But it should work with null too. I am not a big fan of true/false/null for boolean value, but that is maybe me :)
I think the most important thing is that someone who takes the last version has not to think about that if they don't use application gateway, which seems to be the case.
Then, the local variable is not useful indeed, you can directly check the input variable. It avoid to add a local just for that.
Looks good to me!

Reply:
Having another look in the existing variables in variables.tf, I see all of the boolean one are initialized to `true` or `false`. Maybe we should follow the same here? WDYT?

Reply:
My concern is though we set the default value to `false`, the users will see an update-in-place after they upgrade the module's version because only `true`/`false` cannot omit this dynamic block, and I think it'll surprise them.

If you think it's ok I'm happy to set the default value to `false` and remove the dynamic.

Reply:
Not sure to understand how the following would be different:
```hcl
dynamic "ingress_application_gateway" {
      for_each = var.enable_ingress_application_gateway == false ? [] : ["ingress_application_gateway"]
      content {
        enabled      = var.enable_ingress_application_gateway
        gateway_id   = var.ingress_application_gateway_id
        gateway_name = var.ingress_application_gateway_name
        subnet_cidr  = var.ingress_application_gateway_subnet_cidr
        subnet_id    = var.ingress_application_gateway_subnet_id
      }
    }
```
What I am missing?

Reply:
I've tried this way but unfortunately, it'll cause an issue by the following steps:

1. Create a cluster with `var.enable_ingress_application_gateway` as `true`.
2. Change `var.enable_ingress_application_gateway` to `false`.

What we expect is that the cluster is updated to disable ingress application gateway, but the whole `ingress_application_gateway` block will be deleted by `for_each`, in provider's code, this argument is set to `nil`, and this nil value will be passed to the parameter to the API. Azure API will treat a nil field in the patch request as "Do Nothing", so we actually block users from turning off the ingress application gateway.

With default value to `null` and checking null condition for `for_each`, we can turn off the gateway by setting `var.enable_ingress_application_gateway` to `false`.

Reply:
OK, got it! Thanks

Reply:
If you think this pr's ok to merge, I'll ask Nanxuan to approve and merge, and release a new version.

Thank you for your time on this issue! I'm asking Nanxuan how we proceed with the maintenance of these modules, I've seen a lot of pending prs and issues, some of them are easy to implement.

Reply:
Yes, this looks good to me.